source/common/network/proxy_protocol_filter_state.cc

```diff
@@ -5,8 +5,12 @@
 namespace Envoy {
 namespace Network {
 
-const std::string& ProxyProtocolFilterState::key() {
-  CONSTRUCT_ON_FIRST_USE(std::string, "envoy.network.proxy_protocol_options");
+auto proxy_protocol_options_inline_key =
```

Comment:
no global variables. construct_on_first_use at a minimum.

Reply:
I think this is necessary because we will create the inline map objects beofre the the method `key()`. So, `construct_on_first_use` is too late. This is not restriction to the `InlineMap`, but is restriction to the `InlineMap` usage in the `Envoy`. I think all custom HTTP inline header also have same restriction.

The registry used here will be created by the `construct_on_first_use`. So, these definitions have no static initialization order problem.
These inline keys have no non-trivial destructor, so these definitions have no static de-initialization order problem.

Although I know this is not good pattern, i pretty sure it's safe. 
Or we can create a macro to do this definition. At least this will remind other developers not to abuse it.

Reply:
It is not acceptable to have a static-init fiasco. We need to engineer around this.

Reply:
Except the static definition, I still couldn't found an althernative to support the corss-modules registration. orz.

I think this a little like `REGISTER_FACTORY`?